_includes/seo.html

```diff
@@ -4,13 +4,20 @@
 {%- endif -%}
 {%- assign seo_url = seo_url | default: site.github.url -%}
 
-{% assign title_separator = site.title_separator | default: '-' | replace: '|', '&#124;' %}
 
-{%- if page.title -%}
-  {%- assign seo_title = page.title | append: " " | append: title_separator | append: " " | append: site.title -%}
+{%- if site.title_separator -%}
+  {%- assign title_separator = site.title_separator | default: '-' | replace: '|', '&#124;' -%}
+  {%- assign append_title = append: " " | append: title_separator | append: " " | append: site.title -%}
```

Comment:
This smells like a syntax error. Basically you need something to filter, not starting with a filter. 

```diff
-  {%- assign append_title = append: " " | append: title_separator | append: " " | append: site.title -%}
+  {%- assign append_title = " " | append: title_separator | append: " " | append: site.title -%}
```